### PR TITLE
Active Treatment Summary in Overview

### PR DESCRIPTION
**Pull request by Dtphelan1**
Addresses JIRA 1771 - Show active treatment summary in summary section. Introduces a new type of object called `XActiveTreatmentSummary`, which can produce for a given patient and condition an object that has all relevant information regarding that patient's current active treatment for that condition. 

This object is generated via an object factory that looks at the current patient and the condition to determine the kind of active summary object you are interested in; this is currently done via and `instanceof` check on the condition, but in the future one could imagine the selection of XActiveTreatmentSummary objects using a more complex algorithm that inspects the patient for comorbidities, that looks for instances of a more specific class before defaulting to a generalCancerTreatmentSummary, etc. 

Right now the solidTumorActiveTreatmentSummary follows an algorithm I sketched out with @gregquinn2001. I will post an image of the flowchart depicting that decision process on the wiki and link that here shortly. This algorithm is explained at detail in the code, and is only meant to be an initial sketch; as we receive clinician feedback and the develop of more complex patient records, we will expand this algorithm to perform more complex checks for things like previous/schedule procedures (e.g. radiation treatment, rounds of chemo, etc) and to return objects with even more information (e.g. lists of procedures, relevant dates, etc.) 

Last detail: once that informatino is created from the object, the active treatment summary section concatenates relevant information into a human-readable string that can be displayed in the tabular and narrative sections. In the future, this string will be modified to use additional information added to the returned objects, and additional information on the objects it already has. For and example of the second case, we can envision inferring how long the patient has been receiving medication from the start and end dates on the medication object. 

_Pull requests into Flux require the following. Submitter should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:. Reviewers should complete every item on the bulleted list when reviewing._

## Submitter:

**Running the Application**

- [x] Manually tested in Chrome (primary browser for testing)
- [x] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [x] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [x] Branch implements task as expected
- [x] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [x] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [x] This pull request describes why these changes were made
- [x] Recognizes any potential shortcomings/bugs in the description 
- [x] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [x] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [x] Document any new APIs/extension points
- [x] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [x] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [x] Code is commented

**Tests**

- [x] Existing tests passed
- [ ] Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

**Comment by Dtphelan1** [comment]
It's already been pointed out that the naming of these files and the wrappers they reference, so I'll take care of that when the time comes 👍

File: src/patient/PatientRecord.jsx [hunk 1]
```diff
@@ -812,6 +823,31 @@ class PatientRecord {
...
+            // Looking for Surgery code - should be based on http://ncimeta.nci.nih.govs
+            return cond.code === "C0851238";
+        });
+    }
+
+    getSurgeriesPlannedForCondition(condition) {
+        const surgeriesForCondition = this.getSurgeriesForCondition(condition);
+        return surgeriesForCondition.filter((cond) => {
+            // Looking for active request status code - should be based on http://hl7.org/fhir/STU3/valueset-request-status.html
+            return cond.status === "active";
```
Comment (gregquinn2001) [inline src/patient/PatientRecord.jsx]: I'm not sure status is adequate for determining planned surgeries. I think we should be checking the expected performance date to see if it's after today

File: src/patient/PatientRecord.jsx [hunk 2]
```diff
@@ -812,6 +823,31 @@ class PatientRecord {
...
+        })
+    }
+
+    getSurgeriesPreviouslyPerformedForCondition(condition) {
+        const surgeriesForCondition = this.getSurgeriesForCondition(condition);
+        return surgeriesForCondition.filter((cond) => {
+            // Looking for completed request status code - should be based on http://hl7.org/fhir/STU3/valueset-request-status.html
+            return cond.status === "completed";
```
Comment (gregquinn2001) [inline src/patient/PatientRecord.jsx]: Same here. We should be looking for ProcedureRequested with expected performance date in past or really a ProcedurePerformed eventually...

**Comment by Dtphelan1** [comment]
Should address GQ1/2!

File: src/patient/PatientRecord.jsx [hunk 3]
```diff
@@ -812,6 +823,47 @@ class PatientRecord {
             return p.reasons && p.reasons.some(r => r.value.entryId && r.value.entryId === conditionEntryId);
         });
     }
+    
+
+    getSurgeriesForCondition(condition) {
+        const allProceduresForCondition = this.getProceduresForCondition(condition);
+        return allProceduresForCondition.filter((cond) => {
+            // Looking for Surgery code - should be based on http://ncimeta.nci.nih.govs
+            return cond.code === "C0851238";
```
Comment (gregquinn2001) [inline src/patient/PatientRecord.jsx]: Oh sorry I wasn't clear. We should really check the code system uri as well

File: src/summary/activeTreatmentSummary/IActiveTreatmentSummary.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class IActiveTreatmentSummary { 
+    getActiveTreatmentSummary() {
```
Comment (gregquinn2001) [inline src/summary/activeTreatmentSummary/IActiveTreatmentSummary.js]: [GQ4] should have arguments in interface right?

**Comment by Dtphelan1** [comment]
GQ4 addressed!

GQ3 might require additional work because in 0.1 Status codes ('completed', 'active') weren't captured as codeable concepts; information about the codesystem can be inferred during the mapping from 0.1 -> 0.5, but the codesystem is not currently available for Status. As for Surgery, we can check that directly. Implementing now.

**Comment by gregquinn2001** [comment]
For GQ3 status, I think inference is fine in the 0.1 to 0.5 translation code since the status value set only comes from one code system, right?

**Comment by Dtphelan1** [comment]
I think so too. I'll work on the addition of an appropriate codesystem during the entry mapping. With this change to how entries are getting mapped, should we regenerate our 0.5 patient record?

**Comment by gregquinn2001** [comment]
Do we need to regenerate it or just add the code system to the status codes?

**Comment by gregquinn2001** [comment]
For each clinical status for each condition in 0.5 patient, the CodeSystem should be set to http://terminology.hl7.org/CodeSystem/condition-clinical. The entry mapper should do the same for all clinical status values coming from 0.1. Does that sound right?

**Comment by Dtphelan1** [comment]
So in response to your two comments: 
1. I guess we don't have to regenerate it, but there will be a lot of manual replacing since this applies for any procedures in our 0.5 patient. It probably would take less time to chug those changes out than it would to discuss the merits of doing it one way vs the other. 
2. The changes are for `Status`, not `ClinicalStatus`. These are fields that describe the status of a requested 'thing' procedure, medication, etc.). In previous places where we use Status we never included a codesystem, so I've inferred through 'research' (read: rapid googling) that the CodeSystem we probably were implicitly using was http://hl7.org/fhir/STU3/valueset-request-status.html . Let me know if that make sense.

**Comment by gregquinn2001** [comment]
Ok sounds good! Sorry for the confusion on the attribute name. There are more status values than clinical status values. Your plan sounds good!

**Comment by Dtphelan1** [comment]
Not at all 👍 I'll let you know when I've made those changes!

**Comment by Dtphelan1** [comment]
GQ3 should be addressed 👍

**Comment by sgober** [comment]
> It's already been pointed out that the naming of these files and the wrappers they reference will have to change once #613 gets merged, so I'll take care of that when the time comes 👍

Have you handled this yet @Dtphelan1?

**Comment by Dtphelan1** [comment]
Seeing as the branch hasn't been rebased yet, no I havent. Just got back from lunch so doing this now.

**Comment by Dtphelan1** [comment]
Rebased and modified to align with Nicole and Dylan-H's changes. Let me know what I broke :

**Comment by gregquinn2001** [comment]
I was just about to report the path problem and failing test!

**Comment by gregquinn2001** [comment]
FYI. I added active treatment summary to BreastCancerMetadata and got this for active treatment on demo2: Adjuvant PACLitaxel & Lisinopril 20mg tablet

I think that looks right. The data on demo2 isn't the best.

**Pull request merged**